src/main/java/Domain/ShopController.java

```diff
@@ -38,18 +38,42 @@ public Boolean isShopIdExist(Integer shopId)
         return false;
     }
 
-    public void OpenNewShop(Integer shopId, String userName) throws Exception 
+    public void openNewShop(Integer shopId, String userName) throws Exception 
     {
         if(isShopIdExist(shopId))
             throw new Exception(String.format("Shop ID: %d is already exist.", shopId));
         else
             _shopsList.add(new Shop(shopId, userName));
     }
 
-    public void addProductToShop(Integer shopId, Product product, String userName )
+    //close shop only if the user is the founder of the shop
+    public void closeShop(Integer shopId, String userName) throws Exception 
     {
-        //TODO: need to implement  (check if shop exist)
+        try
+        {
+            if(!isShopIdExist(shopId))
+            throw new Exception(String.format("Shop ID: %d does not exist.", shopId));
+            else
+            {
+                Shop shopToClose = getShopByShopId(shopId);
+                if(shopToClose.checkPermission(userName, FOUNDER))
+                    _shopsList.remove(shopToClose);
+            } 
+        }
+        catch(Exception e)
+        {
+            throw new Exception(e.getMessage());
+        }
+   
+    }
+    
 
+    public void addProductToShop(Integer shopId, Product product, String userName) throws Exception
```

Comment:
why in closeShop you check the pernission in the controller and in addProductToShop you check in the shop class?